src/sentry/incidents/endpoints/organization_alert_rule_available_action_index.py

```diff
@@ -65,19 +62,31 @@ def get(self, request, organization):
 
         actions = []
 
-        integrations = get_available_action_integrations_for_org(organization).order_by("id")
+        # Cache Integration objects in this data structure to save DB calls.
         provider_integrations = defaultdict(list)
-        for integration in integrations:
+        for integration in get_available_action_integrations_for_org(organization):
             provider_integrations[integration.provider].append(integration)
-        registered_types = AlertRuleTriggerAction.get_registered_types()
-        registered_types.sort(key=lambda x: x.slug)
 
         for registered_type in AlertRuleTriggerAction.get_registered_types():
+            # Used cached integrations for each `registered_type` instead of making N calls.
             if registered_type.integration_provider:
                 for integration in provider_integrations[registered_type.integration_provider]:
                     actions.append(
                         self.build_action_response(organization, registered_type, integration)
                     )
+
+            # Add all alertable Sentry Apps to the list.
+            elif registered_type.slug == "sentry_app":
```

Comment:
We could use `registered_type.type == AlertRuleTriggerAction.Type.SENTRY_APP` here and avoid hardcoding the string.

Reply:
@mgaeta updates here?

Reply:
I updated the condition here: https://github.com/getsentry/sentry/pull/20263/files#diff-7fba3c7130f3d8dca2c6ab678dc1b488R84